src/components/shared/chart/Canvas.js

```diff
@@ -117,6 +111,18 @@ export default class ChartCanvas<HoveredItem> extends React.Component<
     return this.props.hitTest(x, y);
   }
 
+  _onMouseDown(event: SyntheticMouseEvent<>) {
+    if (!this._canvas || !this.props.onMouseDown) {
+      return;
+    }
+
+    const maybeHoveredItem =
+      this.state.hoveredItem === null
+        ? this._hoveredItemFromMouseEvent(event)
+        : this.state.hoveredItem;
+    this.props.onMouseDown(maybeHoveredItem);
```

Comment:
This is the same solution as for `_onDoubleClick`.
`componentWillReceiveProps` will clear `this.state.hoveredItem` and this makes sure we'll still get the hovered item if the state was unnecessarily cleared.

As soon as `_onMouseMove` is run, we'll get back the hovered item into the state again.